src/Features/CSharp/Portable/ValidateFormatString/ValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ValidateFormatString;
+
+namespace Microsoft.CodeAnalysis.CSharp.ValidateFormatString
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class ValidateFormatStringDiagnosticAnalyzer : AbstractValidateFormatStringDiagnosticAnalyzer
+    {
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpr = (InvocationExpressionSyntax)context.Node;
+            if (invocationExpr.ArgumentList?.Arguments == null)
+            {
+                return;
+            }
+
+            SimpleNameSyntax name;
+            // When calling string.Format(...), Format will be of type MemberAccessExpressionSyntax 
+            if (invocationExpr.Expression is MemberAccessExpressionSyntax memberAccessExpr)
+            {
+                if (memberAccessExpr.Name?.ToString() != "Format")
+                {
+                    return;
+                }
+
+                name = memberAccessExpr.Name;
+            }
+            // When using static System.String and calling Format(...), Format will be of type IdentifierNameSyntax
+            else if (invocationExpr.Expression is IdentifierNameSyntax identifierNameSyntax)
+            {
+                if (identifierNameSyntax.ToString() != "Format")
+                {
+                    return;
+                }
+
+                name = identifierNameSyntax;
+            }
+            else
+            {
+                return;
+            }
+
+            var arguments = invocationExpr.ArgumentList.Arguments;
+            var numberOfArguments = arguments.Count;
+
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(name, context.CancellationToken);
+           
+            if (!TryGetFormatMethod(context, numberOfArguments, symbolInfo, out var method))
+            {
+                return;
+            }
+
+            var hasIFormatProvider =
+                method.Parameters[0].Type.ToString().Equals(typeof(IFormatProvider).FullName);
```

Comment:
What is `FullName` here?  Is it an assembly qualified name?  If so, does this work when running in a PCL/.NETCore system where IFormatProvider might be in a different assembly?

I would tend to do something like `Compilation.GetTypeByMetadataName` to find IFormatProvider.